File: android/app/src/main/java/com/emops/app/domain/model/DsaaLog.kt
package com.emops.app.domain.model

data class DsaaLog(
    val id: String = "",
    val userId: String = "",
    val sheetId: String? = null,
    val logDate: String = "",
    val frictionPoint: String = "",
    val dsaaAction: String = "",
    val microArtifactType: String? = null,
    val microArtifactDescription: String? = null,
    val expectedLeverage: String? = null,
    val startedAt: String? = null,
    val completedAt: String? = null,
    val durationMinutes: Int? = null,
    val aiSuggestedAction: String? = null,
    val aiSuggestionAccepted: Boolean = false
)

data class DsaaSuggestion(
    val dsaaAction: String = "",
    val actionDescription: String = "",
    val microArtifact: String = "",
    val teamMessage: String = "",
    val expectedLeverage: String = ""
)

File: ios/EMOps/Models/DsaaLog.swift
import Foundation

struct DsaaLog: Codable, Identifiable, Equatable {
    let id: String
    let userId: String
    let sheetId: String?
    let logDate: String
    let frictionPoint: String
    let dsaaAction: DsaaAction
    let microArtifactType: String
    let microArtifactDescription: String
    let expectedLeverage: String
    let startedAt: String?
    let completedAt: String?
    let durationMinutes: Int?
    let aiSuggestedAction: String?
    let aiSuggestionAccepted: Bool

    enum CodingKeys: String, CodingKey {
        case id
        case userId = "user_id"
        case sheetId = "sheet_id"
        case logDate = "log_date"
        case frictionPoint = "friction_point"
        case dsaaAction = "dsaa_action"
        case microArtifactType = "micro_artifact_type"
        case microArtifactDescription = "micro_artifact_description"
        case expectedLeverage = "expected_leverage"
        case startedAt = "started_at"
        case completedAt = "completed_at"
        case durationMinutes = "duration_minutes"
        case aiSuggestedAction = "ai_suggested_action"
        case aiSuggestionAccepted = "ai_suggestion_accepted"
    }

    enum DsaaAction: String, Codable, CaseIterable {
        case delete
        case simplify
        case accelerate
        case automate
    }
}
